### PR TITLE
Closes #133 Use languageCode instead of locale identifier for product projection search

### LINKED ISSUE #133: Use languageCode instead of locale identifier for product projection search

### PR DESCRIPTION
**Pull request by nikola-mladenovic**
Found this one out while implementing the localized voice search in Sunrise.

### CODE REVIEW

**Comment by nikola-mladenovic** [comment]
Added searchLanguage for locale method to product projection endpoint

File: Source/ProductProjection.swift [hunk 1]
```diff
@@ -288,4 +284,25 @@ open class ProductProjection: QueryEndpoint, ByIdEndpoint, ByKeyEndpoint, Mappab
...
+        }
+
+        let localeIdentifier = locale.identifier.replacingOccurrences(of: "_", with: "-")
+        if projectLanguages?.contains(localeIdentifier) == true {
+            return localeIdentifier
+        } else if let languageCode = locale.languageCode, projectLanguages?.contains(languageCode) == true {
+            return languageCode
+        } else {
+            return projectLanguages?.first ?? locale.identifier
+        }
```
Comment (cneijenhuis) [inline Source/ProductProjection.swift]: We should not check for string equality, but check whether the project language starts with that string.

As an example, if a project defines `de-DE, en-US` and the user has `en-CA`, the code as it is now wouldn't match in the second if-block and return `de-DE`.
Another example, if a project defines `en-US, zh-Hant, zh-Hans` and the user has `zh-Hans-SG`, the code as it is now wouldn't match in the first or second if-block.

Comment (nikola-mladenovic) [inline Source/ProductProjection.swift]: 👍  Oversight on my end - updated the method.

**Comment by nikola-mladenovic** [comment]
_Note: suggestions method is now using the same logic for setting the language parameter._

**Pull request merged**